### DISCUSSION TITLE
prior distributions for subject specific offsets

### DISCUSSION THREAD

**Discussion by jainraj** [lnccbrown/HSSM#403] [Q&A]
I am trying to replicate the HDDM's Bayesian hierarchical model (see below image - from https://hddm.readthedocs.io/en/latest/methods.html)

<img width="627" alt="Screenshot 2024-04-30 at 14 05 06" src="https://github.com/lnccbrown/HSSM/assets/44366505/dfc015f7-f8df-4672-8918-2d497c25683e">

Is the following the correct configuration for the boundary parameter?

```
{
    "name": "a",
    "formula": "a ~ 1 + (1|participant_id)",
    "link": "identity",
    "prior": {
        "Intercept": {
            "name": "Gamma",
            "mu": 1.5,
            "sigma": 0.75,
        },
        "1|participant_id": {
            "name": "Gamma",
            "mu": 0,
            "sigma": {
                "name": "HalfNormal",
                "sigma": 0.1,
            },
        },
    },
    "bounds": (0.0, numpy.inf),  
},
```

In the generated graph I see following nodes: 
- a_1|participant_id_offset ~ Normal - what exactly is this? 
- a_1|participant_id_sigma ~ HalfNormal 
- a_1|participant_id ~ Deterministic connected to _offset and _sigma 
- a_Intercept ~ Gamma 

And when I look at plot_priors for a_1|participant_id - they all look Gaussian - Shouldn't they be Gamma distributed?. Seems like I am missing something basic. 

PS: It is my first time using Hierarchical Bayesian models

**Comment by AlexanderFengler** [comment]
Hey @jainraj ,

I think the correct way to specify exactly as in HDDM for `a` would be to use `a ~ 0 + (1|participant_id)`, with, 

```
"1|participant_id": {
            "name": "Gamma",
            "mu": {"name": "Gamma",
                   "mu": 1.5,
                   "sigma": 0.75,
                  }
,
            "sigma": {
                "name": "HalfNormal",
                "sigma": 0.1,
            },
        },
```

Best,
Alex

**Reply by jainraj** [reply]
So @AlexanderFengler, 

1. Does the `0` indicate that we are not learning the common offset, i.e., not a mixed-effects model (which HDDM couldn't do)? 
2. The model graph shows up like this, which is a bit weird. Why isn't the "mu" connected to any node? Is it a plotting problem or does it actually mean that this parameter won't be tuned?
 
![discussion](https://github.com/lnccbrown/HSSM/assets/44366505/100a2de8-029f-4d0e-bb29-d5d2074e73ca)

**Reply by jainraj** [reply]
So, I ran a simple model with this version, and it seems like the `mu` parameter does get tuned. Makes me think that this is more of a plotting problem. @AlexanderFengler if you can confirm this, maybe this can be taken as a bug and discussion can be closed?
